src/utils/options_utils.pas

```diff
@@ -220,6 +228,47 @@ procedure FontToSetOptions(AFont: TFont; const FontName, SizeName, ColorName,
     ChangeSetOption(SetOptions[StyleName], StyleName, SetToString(PTypeInfo(TypeInfo(TFontStyles)), Integer(AFont.Style), false), rtString);
 end;
 
+function ChartColorsFromOptions(OptionList: TOptionList; SetOptions: TSetOptionsMap = nil): TColorMap;
+var
+  i:    Integer;
+  iMap: Integer;
+  vMap: TDigitIndex;
+  Opt:  TOption;
+  anaColors: array of TColor = (clBlue, clRed, clBlack, clGreen, clYellow, clWhite, clSkyBlue, clFuchsia, clGray, clAqua);
+begin
+  result := copy(anaColors);
+  if (Assigned(SetOptions)) then
+    begin
+      iMap := StrToInt(SetOptions.GetValue(ANA_SO_CHART_COLORS).Value);
+    end;
+  if (OptionList.HasOption('colors', Opt)) then
+    iMap := Opt.Expr.AsInteger;
```

Comment:
I thought to have the option validated by ast - allowing only an integer.
I see that this probably is not a good idea, since most set options are strings and colours in Classic used a string.
I've fixed this and have the ChartColors... utility validate colours (set to colour '0' anything that is not a digit). For now, there is no warning to the user. The documentation can caution about only using digits, I think.